container/docker/factory.go

```diff
@@ -140,6 +145,12 @@ func FullContainerName(dockerId string) string {
 func (self *dockerFactory) CanHandleAndAccept(name string) (bool, bool, error) {
 	// docker factory accepts all containers it can handle.
 	canAccept := true
+
+	// When using Systemd all docker containers have a .scope suffix
+	if UseSystemd() && !strings.HasSuffix(path.Base(name), ".scope") {
```

Comment:
Feels appropriate to follow existing format and express the check under `isContainerName` or similar function?